### PR TITLE
add log/root-det tests and duals, and support duals through det bridge

### PR DESCRIPTION
**Pull request by chriscoey**

### CODE REVIEW

File: test/Test/contconic.jl [hunk 1]
```diff
@@ -216,14 +216,53 @@ end
...
-
-    mock.optimize! = (mock::MOIU.MockOptimizer) -> MOIU.mock_optimize!(mock, [1, 1, 0, 1])
-    MOIT.rootdetttest(mock, config)
-    mock.optimize! = (mock::MOIU.MockOptimizer) -> MOIU.mock_optimize!(mock, [1, 1, 0, 0, 1])
-    MOIT.rootdetstest(mock, config)
+@testset "LogDet" begin
+    # TODO fix: errors with "Fallback getter for variable constraint dual does not support other variable-wise constraints on the variable"
+    # mock.optimize! = (mock::MOIU.MockOptimizer) -> MOIU.mock_optimize!(mock, [0, 1, 0, 1, 1],
+    #     (MOI.SingleVariable, MOI.EqualTo{Float64}) => [2],
+    #     (MOI.VectorAffineFunction{Float64}, MOI.Nonnegatives) => [[1, 1]])
```
Comment (blegat) [inline test/Test/contconic.jl]: You might need to set the dual for the VOV in LogDet constraint

Comment (chriscoey) [reply test/Test/contconic.jl]: the following triggers the same error: 
```
    mock.optimize! = (mock::MOIU.MockOptimizer) -> MOIU.mock_optimize!(mock, [0, 1, 0, 1, 1],
        (MOI.SingleVariable, MOI.EqualTo{Float64}) => [2],
        (MOI.VectorAffineFunction{Float64}, MOI.Nonnegatives) => [[1, 1]],
        (MOI.VectorOfVariables, MOI.LogDetConeTriangle) => [[-1, -2, 1, 0, 1]])
    MOIT.logdett1vtest(mock, config)
```
error:
```
LogDet: Error During Test at /home/coey/.julia/dev/MathOptInterface/src/Test/contconic.jl:2689
  Test threw exception
  Expression: ≈(MOI.get(model, MOI.ConstraintDual(), vc), 2, atol = atol, rtol = rtol)
  Fallback getter for variable constraint dual does not support other variable-wise constraints on the variable. Please report this issue to the solver wrapper package.
  Stacktrace:
   [1] error(::String, ::String, ::String) at ./error.jl:42
   [2] variable_dual(::MathOptInterface.Utilities.MockOptimizer{MathOptInterface.Utilities.Model{Float64}}, ::MathOptInterface.ConstraintDual, ::MathOptInterface.ConstraintIndex{MathOptInterface.SingleVariable,MathOptInterface.EqualTo{Float64}}, ::MathOptInterface.VariableIndex, ::Type{MathOptInterface.VectorOfVariables}, ::Type{MathOptInterface.LogDetConeTriangle}) at /home/coey/.julia/dev/MathOptInterface/src/Utilities/results.jl:331
   [3] variable_dual(::MathOptInterface.Utilities.MockOptimizer{MathOptInterface.Utilities.Model{Float64}}, ::MathOptInterface.ConstraintDual, ::MathOptInterface.ConstraintIndex{MathOptInterface.SingleVariable,MathOptInterface.EqualTo{Float64}}, ::MathOptInterface.VariableIndex) at /home/coey/.julia/dev/MathOptInterface/src/Utilities/results.jl:381
   [4] variable_dual at /home/coey/.julia/dev/MathOptInterface/src/Utilities/results.jl:398 [inlined]
   [5] get_fallback at /home/coey/.julia/dev/MathOptInterface/src/Utilities/results.jl:423 [inlined]
   [6] get at /home/coey/.julia/dev/MathOptInterface/src/Utilities/mockoptimizer.jl:384 [inlined]
   [7] _det1test(::MathOptInterface.Utilities.MockOptimizer{MathOptInterface.Utilities.Model{Float64}}, ::MathOptInterface.Test.TestConfig{Float64}, ::Bool, ::Type{T} where T) at /home/coey/.julia/dev/MathOptInterface/src/Test/contconic.jl:2689
```

Comment (blegat) [reply test/Test/contconic.jl]: Use `eval_variable_constraint_dual = false` in `MockOptimizer` constructor

Comment (chriscoey) [reply test/Test/contconic.jl]: thanks! fixed

**Pull request merged**